_projects/go_bears_on_bsd.md

```diff
@@ -0,0 +1,102 @@
+---
+collaborating_projects:
+  - coala
+desc: "Explore improvements to coala's dependency management through having Go Linter installed in *BSD system automatially."
+difficulty: medium
+initiatives:
+  - GSoC
+issues:
+  - "https://gitlab.com/coala/package_manager/issues/149"
+  - "https://github.com/coala/coala-bears/issues/2481"
+markdown: go_bears_on_bsd.md
+mentors:
+  - xferra
+  - yukiisbored
+name: "Linter Dependency Management from a Bare-Metal System"
+requirements:
+  - "At least one patch to the coala core should be accepted and merged."
+  - "The proposal needs to show familiarity with coala's dependency management implementation."
+status: in_progress
+tags:
+  - Automation
+  - Installation
+  - Dependency
+---
+
+`coala-bears` are of two types, Native Bears which implement all the business
+logic themselves and Linter Bears which wrap existing linters and checkers.
+Dependencies of bears are mostly language runtimes, language-specific packages
+(e.g. python:nltk) or command line tools (e.g. git). Both Native Bears and
+Linter Bears have dependencies which need to be installed before they can be
+executed.
+
+`dependency_management` aims to provide Python APIs for dependency management on
+various languages and different systems. It is used extensively by coala,
+however, it lacks most of the key implementation and documentation in its
+current state.
+
+coala now has [7
+bears](https://github.com/coala/coala-bears/blob/master/bears/go/) for the Go
+language. They need specific versions of Go and some externally supplied
+packages, and currently Go linter bears just assume they are all in place and
+work. Go language dependencies are currently managed unversioned ("If you're
+using an externally supplied package and worry that it might change in
+unexpected ways, the simplest solution is to copy it to your local repository.
+This is the approach Google takes internally", see [Go Official
+FAQ](https://golang.org/doc/faq#get_version)).
```

Comment:
while this is the Google official position, I understand there are/were other commonly used mechanisms for managing to install specific versions of go repos.  branches was one of them.

This doesnt need to be addressed in the first iteration of this project spec, but I believe it should be covered in detail in a spec or in our documentation before this project attempts installing various versions.